### ISSUE TITLE
Can't delete from gmail

### ISSUE THREAD

**Issue by xgvargas** [Foundry376/Mailspring#1513]
##### What operating system are you using?
linux mint 19
...

##### What version of Mailspring are you using?
1.6.1-aceed8e5
...

**Bug?**

##### Is this an email sync or connection issue?
no
...

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
gmail
...

##### Is the issue reproducible with a particular attachment, message, signature, etc?

I do have two accounts, one `office 356` that works 100%, and one `gmail` that receive and send email, also archive them, but the delete/trash button does nothing. Not event to email being composed the trash button works. It always fail silently. But if I open the developer tools the console shows the following error every time I press the button:

```
/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:21 Uncaught Error: ChangeFolderTask: You must provide a single folder.
    at new ChangeFolderTask (change-folder-task.ts:70)
    at tasksForThreadsByAccountId (task-factory.ts:88)
    at Object.values.forEach (task-factory.ts:29)
    at Array.forEach (<anonymous>)
    at Object.tasksForThreadsByAccountId (task-factory.ts:28)
    at Object.tasksForMovingToTrash (task-factory.ts:87)
    at TrashButton._onRemove (thread-toolbar-buttons.tsx:69)
    at Object.fa (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:14)
    at ma (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:15)
    at na (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:15)
    at za (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:17)
    at Da (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:18)
    at Fa (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:18)
    at Ba (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:18)
    at Ia (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:21)
    at be (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:84)
    at Ii (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:231)
    at bc (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:39)
    at ee (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:86)
    at Ki (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:232)
    at de (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:85)
```
`Must provide a single folder`.... where?

If I go to `preferences > folders` I see my gmail account has `trash` pointing to `deleted items`, but the `spam` was `(none)`, which was different from the other account the works. So tried to set to `deleted messages` but the error remains.

Nobody seems to be having this issue so i think I may be missing something... any ideas?

**Comment by Hype-Rat** [comment]
It looks like the **node_modules** are giving this error. I will take a look :)

**Comment by xgvargas** [comment]
Gave a new try with version `1.6.3` just to get the very same error.
The difference is that this time the only option available for trash folder is `Trash` and `[GMail/All emails]`, but if I select the later it reverts to the former. And the error remains: 

```
/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:21 Uncaught Error: ChangeFolderTask: You must provide a single folder.
    at new ChangeFolderTask (change-folder-task.ts:72)
    at tasksForThreadsByAccountId (task-factory.ts:84)
    at Object.values.forEach (task-factory.ts:29)
    at Array.forEach (<anonymous>)
    at Object.tasksForThreadsByAccountId (task-factory.ts:28)
    at Object.tasksForMovingToTrash (task-factory.ts:81)
    at TrashButton._onRemove (thread-toolbar-buttons.tsx:69)
    at Object.fa (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:14)
    at ma (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:15)
    at na (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:15)
    at za (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:17)
    at Da (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:18)
    at Fa (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:18)
    at Ba (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:18)
    at Ia (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:21)
    at be (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:84)
    at Ii (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:231)
    at bc (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:39)
    at ee (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:86)
    at Ki (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:232)
    at de (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:85)
```

**Comment by xgvargas** [comment]
Every once in a while I give mailspring a try due to my hate of thunderbird (no specific reason, just pure true unconditional hate). And, for some reason I always got the errors listed above when trying mailspring. Thought it was an client bug. 

But this time I put some more effort to fix the issue and finally it is working.

The issue was on my gmail configuration. Since I've installed a lot of email clients and most of them have created their own markers (mailsprint itself have created 2 markers in there), somehow I had 2 markers named trash and both of them were imap enabled (yes, i'm using imap as part of my effort to fix the original issue). As soon I got them deleted, mailsprint started working properlly.

Maybe the error message could be improoved to make debuging of this situation easier, but any way, it's working and mailspring is great!
